### PR TITLE
Upgrade to Mypy 1.0

### PR DESCRIPTION
**Pull request by christianbundy**
# I have made things!

Upgrade to Mypy 1.0

## Related issues

Refs https://github.com/python/mypy/issues/13685

### CODE REVIEW

**Comment by intgr** [comment]
Awesome. Is there a reason this is still in draft?

**Comment by christianbundy** [comment]
Nope! Just didn't notice that CI passed. ✅

File: setup.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ def find_stub_files(name: str) -> List[str]:
 ]
 
 extras_require = {
-    "compatible-mypy": ["mypy>=0.991,<1.0"],
+    "compatible-mypy": ["mypy>=1.0,<1.1"],
```
Comment (ljodal) [inline setup.py:34]: Do we need to bump the minimum, or is that the policy? Seems like there's no braking changes here?

Comment (intgr) [reply setup.py:34]: This only affects those that use `django-stubs[compatible-mypy]`, yes we bump lower bound in that case.

Comment (ljodal) [reply setup.py:34]: Ah right, nvm me, I'm blind 😇

**Review by intgr — APPROVED** [review]
Looks fine to me.

**Review by sobolevn — APPROVED** [review]
LGTM! Thanks a lot!

**Comment by adamchainz** [comment]
Nice one!

**Comment by noamcohen97** [comment]
Nice! Is there a new release planned?

**Comment by intgr** [comment]
@noamcohen97 See #1363

**Pull request merged**